### ISSUE TITLE
Potential issue with generic classes

### ISSUE THREAD

**Issue by rogerlord** [tom-englert/Lazy.Fody#7]
Hi there - recently started looking into AOP to clean up some calculations that are backed by lazy fields.
Lazy.Fody seems to work great for most of them, but I'm not sure it works for generic classes. A simplified example can be found below:

```
public class Calculator<T>
{
    [Lazy]
    public double Calculation => 3.14;
}

public class CalculatorTests
{
    [Fact]
    public void Test()
    {
        var calculator = new Calculator<double>();
        var result = calculator.Calculation;
        Assert.Equal(3.14, result);
    }
}
```

Running this unit test gives the following exception:

`System.InvalidOperationException: Could not execute the method because either the method itself or the containing type is not fully instantiated.`

Is this a known issue? If so - any suggestions where to start looking for a fix?

**Comment by rogerlord** [comment]
I obviously agree the use of generics in this example is completely pointless - but that's just to generate a small concise example in which Lazy.Fody seems to fail. In the real-life example the generic parameter is used in the class.

**Comment by tom-englert** [comment]
Just did not consider this use case yet. Fixed in V 1.11

**Comment by rogerlord** [comment]
Many thanks - works like a charm!
